### ISSUE TITLE
Integration with Angular-CLI

### ISSUE THREAD

**Issue by mb-jenks** [angular-redux/ng2-redux#113]
Is there a way to provide a `/dist` folder for integrating with angular-cli projects?  Because of the mapping associated with the angular-cli build, its really difficult to use the ng2-redux package, specifically its dependency on `lodash` and the mapping of `rxjs` within the src files (the angular-cli uses `rxjs/Rx`).

**Comment by SethDavenport** [comment]
@mb-jenks I've only just started playing with angular-cli myself, but it's definitely something we should be able to work with.  We'll look into it.

**Comment by SethDavenport** [comment]
With PR #115 , I'm able to generate a `lib` folder that works with a simple angular-cli app I wrote. The other secret sauce is to add the following configs to your Angular-CLI-generated app (see the [angular-cli docs](https://github.com/angular/angular-cli/wiki/3rd-party-libs)):

in `angular-cli-build.js`:

```javascript
/* global require, module */

var Angular2App = require('angular-cli/lib/broccoli/angular2-app');

module.exports = function(defaults) {
  return new Angular2App(defaults, {
    vendorNpmFiles: [
       //...
      'redux/dist/redux.js',
      'ng2-redux/lib/**/*.+(js|js.map)'
    ]
  });
};
```

and in `src/system-config.ts`:

```typescript
const map: any = {
  redux: 'vendor/redux/dist/redux.js',
  'ng2-redux': 'vendor/ng2-redux/lib'
};

/** User packages configuration. */
const packages: any = {
  'ng2-redux': {
    format: 'cjs',
    defaultExtension: 'js',
    main: 'index'
  }
};
```

Ah, the joys of SystemJS I thought I left behind me when we switched to Webpack. Now the CLI is sucking me back into SystemJS' rabbit hole of redundant configs ;)

As an aside, I logged issue angular/angular-cli/#882 against the CLI to see if there are plans to simplify bringing in libs from NPM, we'll see if anything transpires there (* cough * ditching SystemJS * cough *).

**Comment by mb-jenks** [comment]
Thanks @SethDavenport !  Do you know when the changes will be merged with master?  The systemjs configs are a pain but I've got them working for now.

**Comment by SethDavenport** [comment]
Waiting for the go-ahead from @e-schultz

**Comment by e-schultz** [comment]
@mb-jenks @SethDavenport changes are merged in, and a v2.3.4 has just been published on npm.
